core/src/main/scala/org/http4s/multipart/file/MixedPart.scala

```diff
@@ -0,0 +1,85 @@
+package org.http4s
+package multipart
+package file
+
+import cats.effect.Sync
+import fs2.Stream
+import fs2.io.{readInputStream, file => fs2File}
+import fs2.text.utf8Encode
+import java.io.{File, FileInputStream, InputStream}
+import java.net.URL
+import java.nio.file.{Files, Path}
+
+import org.http4s.headers.`Content-Disposition`
+
+sealed abstract class MixedPart[F[_]] {
+  def headers: Headers
+
+  def body(implicit F: Sync[F]): Stream[F, Byte]
+
+  def name: Option[String] = headers.get(`Content-Disposition`).flatMap(_.parameters.get("name"))
+
+  def filename: Option[String] =
+    headers.get(`Content-Disposition`).flatMap(_.parameters.get("filename"))
+}
+
+final case class BasicPart[F[_]](headers: Headers, bodyStream: Stream[F, Byte])
+    extends MixedPart[F] {
+
+  def body(implicit F: Sync[F]): Stream[F, Byte] = bodyStream
+}
+
+final case class FilePart[F[_]](headers: Headers, fileStream: Stream[F, Byte], filePath: Path)
+    extends MixedPart[F] {
+
+  def cleanupFile(implicit F: Sync[F]): F[Unit] =
+    F.delay(Files.delete(filePath))
+
+  def body(implicit F: Sync[F]): Stream[F, Byte] = fileStream.onFinalize(cleanupFile)
```

Comment:
`fileStream` cleans up the file and `body` does not.  Is there any reason to expose the former?